src/Agent.Worker/Release/ReleaseCommandExtension.cs

```diff
@@ -12,76 +12,68 @@
         {
-            if (string.Equals(command.Event, WellKnownReleaseCommand.UpdateReleaseName, StringComparison.OrdinalIgnoreCase))
-            {
-                ProcessReleaseUpdateReleaseNameCommand(context, command.Data);
-            }
-            else
-            {
-                throw new Exception(StringUtil.Loc("RMCommandNotFound", command.Event));
-            }
+            CommandArea = "release";
+            SupportedHostTypes = HostTypes.Release | HostTypes.Deployment;
+            InstallWorkerCommand(new ReleaseUpdateReleaseNameCommand());
         }
 
-        private void ProcessReleaseUpdateReleaseNameCommand(IExecutionContext context, string data)
+        private class ReleaseUpdateReleaseNameCommand: IWorkerCommand
         {
-            ArgUtil.NotNull(context, nameof(context));
-            ArgUtil.NotNull(context.Endpoints, nameof(context.Endpoints));
-
-            Guid projectId = context.Variables.System_TeamProjectId ?? Guid.Empty;
-            ArgUtil.NotEmpty(projectId, nameof(projectId));
-            
-            string releaseId = context.Variables.Release_ReleaseId;
-            ArgUtil.NotNull(releaseId, nameof(releaseId));
+            public string Name => "updatereleasename";
+            public string[] Aliases => new string [] {};
 
-            if (!String.IsNullOrEmpty(data))
+            public void Execute(IExecutionContext context, Command command)
             {
-                // queue async command task to update release name.
-                context.Debug($"Update release name for release: {releaseId} to: {data} at backend.");
-                var commandContext = HostContext.CreateService<IAsyncCommandContext>();
-                commandContext.InitializeCommandContext(context, StringUtil.Loc("RMUpdateReleaseName"));
-                commandContext.Task = UpdateReleaseNameAsync(commandContext,
-                                                             context,
-                                                             WorkerUtilities.GetVssConnection(context),
-                                                             projectId,
-                                                             releaseId,
-                                                             data,
-                                                             context.CancellationToken);
-                context.AsyncCommands.Add(commandContext);
+                string data = command.Data;
+                ArgUtil.NotNull(context, nameof(context));
+                ArgUtil.NotNull(context.Endpoints, nameof(context.Endpoints));
+
+                Guid projectId = context.Variables.System_TeamProjectId ?? Guid.Empty;
+                ArgUtil.NotEmpty(projectId, nameof(projectId));
+                
+                string releaseId = context.Variables.Release_ReleaseId;
+                ArgUtil.NotNull(releaseId, nameof(releaseId));
+
+                if (!String.IsNullOrEmpty(data))
```

Comment:
generally, prefer `string` or `String` type usage? I guess this project tends to use the newer `string`

Reply:
I didn't change any of the code in this method, other than to add the line:
```
string data = command.Data
```
so that the command signature matched